### PR TITLE
fix self link

### PR DESCRIPTION
**Pull request by amy**

### CODE REVIEW

File: service/catalog.go [hunk 1]
```diff
@@ -19,10 +19,18 @@ func getCatalogs(w http.ResponseWriter, r *http.Request, envId string) (int, err
 
 	resp := model.CatalogCollection{}
 	for _, catalog := range catalogs {
+
+		selfLink := apiContext.UrlBuilder.ReferenceByIdLink("catalogs", catalog.Name)
+		projectID := r.URL.Query().Get("projectId")
+		if projectID != "" {
+			selfLink = selfLink + "?projectId=" + projectID
```
Comment (joshwget) [inline service/catalog.go]: Links to catalogs shouldn't need this parameter. I think this can be omitted.

**Comment by amy** [comment]
cc/ @joshwget @cjellick

**Comment by joshwget** [comment]
LGTM

**Comment by cjellick** [comment]
Needs some more. You are only doing this in getCatalogs function (which is mapped to `/v1-catalog/catalogs`, but what about `/v1-catalogs/catalog/my-catalog`? That needs to return a proper self link too.

I suggest having a constructor function for  `model.CatalogResource` and _always* calling that instead of directly creating a new one via `c := &model.CatalogResoure`.

**Comment by cjellick** [comment]
Closing in favor of https://github.com/rancher/catalog-service/pull/80, which is @amy's PR with one tiny tweak to the test.
Just looking to save you some time on having to edit it again @amy.

**Pull request closed without merging**